### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW13 2024)

### PR DESCRIPTION
**Pull request by sys-ce-bb**
LLVM: llvm/llvm-project@acdba090f2724335cf6cf6ecbfb68f67f0cd2def
SPIRV-LLVM-Translator: KhronosGroup/SPIRV-LLVM-Translator@1a1b648cd191eb49d4aa5b95f9ace1c16fc700f2

### CODE REVIEW

**Comment by jsji** [comment]
This is ready for review.
- [x] [Revert "[CodeGen][arm64e] Add methods and data members to Address, wh…](https://github.com/intel/llvm/pull/13190/commits/e2ce547dd35024d41a5e0dba12f740cea6b2a4f7) Cherry-picked revert to fix test failures , no review required.
- [ ] [Fix tbaa.struct overlapping issue found by verifier](https://github.com/intel/llvm/pull/13190/commits/a760456918139f278b70c81b2fd2b674a1ac8fe1)  SPIRV test IR update due to errors exposed by new tbaa verfier. @intel/dpcpp-tools-reviewers  
- [ ] [Update lit due to Target ID difference](https://github.com/intel/llvm/pull/13190/commits/08ce280f909f06d6d490c18915a63771abba459d) Lit test failures due to reverted commits target id unification. @intel/dpcpp-tools-reviewers 
- [ ] [Fix inrange GEP](https://github.com/intel/llvm/pull/13190/commits/12cfa16681c3ae03c9b8c9079829be6a6c804a3d) Fix to workaround for GEP, due to inrange IR change. Already reviewed internally by @asudarsa . @intel/dpcpp-tools-reviewers 
- [ ] [Revert "[RemoveDIs] Load into new debug info format by default in llv…](https://github.com/intel/llvm/pull/13190/commits/023653f549199996b6ceeb7b9be3b2ed43b106af) Temporarily revert , we are not ready to enable new debug info format in llvm-link yet, there are massive failures. Need follow up in https://github.com/intel/llvm/issues/13228

**Comment by jsji** [comment]
@intel/dpcpp-tools-reviewers  Can you please help review and add comments/approvals explicitly to this PR so that this can be merged. Thanks.

**Comment by asudarsa** [comment]
> @intel/dpcpp-tools-reviewers Can you please help review and add comments/approvals explicitly to this PR so that this can be merged. Thanks.

Hi @jsji 
I am looking at the changes now.

Thanks

**Comment by jsji** [comment]
> > @intel/dpcpp-tools-reviewers Can you please help review and add comments/approvals explicitly to this PR so that this can be merged. Thanks.
> 
> Hi @jsji I am looking at the changes now.
> 
> Thanks

Thank you @asudarsa ! Ping me if you need any information about these changes.

**Review by asudarsa — APPROVED** [review]
We might need other changes along with this one. I am not sure how many were brought in through the pulldown. May be we missed some.

Reverting this looks ok.

Thanks

**Comment by jsji** [comment]
> We might need other changes along with this one. I am not sure how many were brought in through the pulldown. May be we missed some.

For sycl branch, we should be in sync with main through previous pulldown (unless I missed something.) 

Yes, revert is temporarily, we should track down and either fix, recover or cherry-pick required commits. Thanks @asudarsa

**Comment by jsji** [comment]
@bader @intel/llvm-gatekeepers This is ready for merge. Thanks.

**Review by asudarsa — COMMENTED** [review]
Do we know why this started failing? I will also check.

Thanks

File: llvm-spirv/test/extensions/INTEL/SPV_INTEL_fpga_reg/IntelFPGAReg.ll [hunk 1]
```diff
@@ -368,7 +368,7 @@ attributes #4 = { nounwind }
 !11 = !{i64 0, i64 4, !9, i64 4, i64 4, !12}
 !12 = !{!13, !13, i64 0}
 !13 = !{!"float", !7, i64 0}
-!14 = !{i64 0, i64 4, !9, i64 0, i64 4, !15}
+!14 = !{i64 0, i64 4, !15}
```
Comment (asudarsa) [inline llvm-spirv/test/extensions/INTEL/SPV_INTEL_fpga_reg/IntelFPGAReg.ll:371]: Do we know why this started failing? I will also check

Comment (jsji) [reply llvm-spirv/test/extensions/INTEL/SPV_INTEL_fpga_reg/IntelFPGAReg.ll:371]: This is exposed by the newly enabled TBAA verfier.
* df75183d70e0 - (2024-03-27) [TBAA] Add verifier for tbaa.struct metadata (#86709)

**Comment by bader** [comment]
https://github.com/intel/llvm/pull/13190/commits/12cfa16681c3ae03c9b8c9079829be6a6c804a3d - @jsji, @asudarsa is this change being pushed to LLVM-SPIRV-Translator repository?

**Comment by jsji** [comment]
> [12cfa16](https://github.com/intel/llvm/commit/12cfa16681c3ae03c9b8c9079829be6a6c804a3d) - @jsji, @asudarsa is this change being pushed to LLVM-SPIRV-Translator repository?

I don't think so, the whole workaround in `simplifyBuiltinVarAccesses` is `sycl` branch specific https://github.com/intel/llvm/pull/10407 , and as mentioned in the comment: "This is a temporary workaround to deal with a graphics driver failure not  able to support the typed pointer reverse translation of getelementptr i8, ptr @__spirv_Builtin* patterns.".

The problem was fixed in linux driver already, but unfortunately, still not fixed yet in Windows driver.

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Mon 01 Apr 2024 04:50:39 PM UTC --- Start to merge the commit into sycl branch. It will take several minutes.

**Comment by bb-sycl** [comment]
Mon 01 Apr 2024 04:55:04 PM UTC --- Merge the branch in this PR to base automatically. Will close the PR later.

**Pull request merged**